packages/slice-machine/src/legacy/lib/models/common/widgets/Link/components.tsx

```diff
@@ -0,0 +1,67 @@
+import { Box, Icon, Tooltip } from "@prismicio/editor-ui";
+import React, { useEffect, useState } from "react";
+import { Checkbox, Flex, Label } from "theme-ui";
+
+import { Col } from "@/legacy/components/Flex";
+
+type DisplayTextModel = { type: string };
+
+interface DisplayTextCheckboxProps {
+  text?: DisplayTextModel;
+  height?: 130 | 127;
+  setFieldValue: (
+    a: string,
+    b?: DisplayTextModel,
+    // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents, @typescript-eslint/no-explicit-any
+  ) => void | Promise<void | any>;
+}
+
+export function DisplayTextCheckbox(props: DisplayTextCheckboxProps) {
+  const { text, height = 130, setFieldValue } = props;
+
+  const [allowDisplayText, setAllowDisplayText] = useState(Boolean(text));
+
+  useEffect(() => {
+    void setFieldValue(
+      "config.text",
+      allowDisplayText ? { type: "Text" } : undefined,
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allowDisplayText]);
+
+  return (
+    <Col>
+      <Flex
+        sx={{
+          mt: 2,
```

Comment:
Same as before, i wonder if we can remove this extra wrapping element and if the spacing should be applied in the parent instead with `gap`.

Reply:
Same comment as my response above:

> Yes, the <Box sx={{ mt: 2 }}> can be avoided by adding a gap to the grid. But it will mean having to update every form for every field.
> ...
> This doesn't seem worth it to me as I believe that we're going to rework all the field forms soon anyway. What do you think?